packages/components/src/test/TestUtils.tsx

```diff
@@ -43,10 +44,14 @@ export async function waitForElement<T extends HTMLElement>(container: HTMLEleme
  *
  * It should be used when test need to do interactions with rendered components.
  */
-function customRender(ui: ReactElement, options?: RenderOptions & { disableStrictMode?: boolean }): RenderResult & { user: UserEvent } {
+function customRender(
+  ui: ReactElement,
+  options?: RenderOptions & { disableStrictMode?: boolean; addThemeProvider?: boolean },
+): RenderResult & { user: UserEvent } {
   const wrapper = createWrapper(options?.wrapper, options?.disableStrictMode);
+
   return {
-    ...renderRTL(ui, { ...options, wrapper }),
+    ...renderRTL(options?.addThemeProvider ? <ThemeProvider includeCss={false}>{ui}</ThemeProvider> : ui, { ...options, wrapper }),
```

Comment:
Setting theme provider for all tests resulted in a lot of failures for different tests as some relied on counting children nodes of rendered components. I thought that it's a good idea to leave these failing tests as is (because they seem perfectly fine) and only change tests that need to be changed.

Reply:
It might be because ThemeProvider was added as component in tree passed to render. Can you try after moving it to options.wrapper?

Reply:
Tried moving, the same issue persists.

Reply:
Ok, we can leave `addThemeProvider` options but I think it still should be moved to `createWrapper` and added through `renderOptions.wrapper`